addons/web_editor/static/src/js/wysiwyg/widgets/image_crop_widget.js

```diff
@@ -1,174 +1,238 @@
+            this.backdrop.remove();
+            this.document.removeEventListener('click', this._onOutsideClick, {capture: true});
         }
-        this._super.apply(this, arguments);
+        return this._super(...arguments);
     },
     /**
      * Updates the DOM image with cropped data and associates required
      * information for a potential future save (where required cropped data
      * attachments will be created).
      *
-     * @override
+     * @private
      */
-    save: function () {
-        var self = this;
-        var cropperData = this.$cropperImage.cropper('getData');
-
-        // Mark the media for later creation of required cropped attachments...
-        this.$media.addClass('o_cropped_img_to_save');
-
-        // ... and attach required data
-        this.$media.data('crop:resModel', this.options.res_model);
-        this.$media.data('crop:resID', this.options.res_id);
-        this.$media.data('crop:id', this.imageData.id);
-        this.$media.data('crop:mimetype', this.imageData.mimetype);
-        this.$media.data('crop:originalSrc', this.imageData.originalSrc);
-
-        // Mark the media with the cropping information which is required for
-        // a future crop edition
-        this.$media
-            .attr('data-aspect-ratio', this.imageData.aspectRatio)
-            .data('aspectRatio', this.imageData.aspectRatio);
-        _.each(cropperData, function (value, key) {
-            key = _.str.dasherize(key);
-            self.$media.attr('data-' + key, value);
-            self.$media.data(key, value);
+    _save() {
+        // Mark the media for later creation/update of cropped attachment
+        // see 'saveCroppedImages' in rte.summernote.js
+        this.media.classList.add('o_cropped_img_to_save');
+
+        this.allAttributes.forEach(attr => {
+            delete this.media.dataset[attr];
+            const value = this._getAttributeValue(attr);
+            if (value) {
+                this.media.dataset[attr] = value;
+            }
         });
 
         // Update the media with base64 source for preview before saving
-        var canvas = this.$cropperImage.cropper('getCroppedCanvas', {
+        const cropperData = this.$cropperImage.cropper('getData');
+        const canvas = this.$cropperImage.cropper('getCroppedCanvas', {
             width: cropperData.width,
             height: cropperData.height,
         });
-        this.$media.attr('src', canvas.toDataURL(this.imageData.mimetype));
-
-        this.final_data = this.media;
-        return this._super.apply(this, arguments);
+        // 1 is the quality if the image is jpeg (in the range O-1), defaults to .92
+        this.media.setAttribute('src', canvas.toDataURL(this.mimetype, 1));
+        this.destroy();
+    },
+    /**
+     * Discards the current crop.
+     *
+     * @private
+     */
+    _discard() {
+        this.media.setAttribute('src', this.initialSrc);
+        return this.destroy();
+    },
```

Comment:
Feels like this should simply be an override of the destroy method. Maybe the `else` part of your condition in there. If the parent of the widget is destroyed, the widget will be destroyed and the initialSrc should be restored, right?